### PR TITLE
Added scripting support

### PR DESCRIPTION
**Pull request by anubhavgoel26**
Adds scripts for graphing and restructuring data the scripts folder:
- divide_into_batches.py: takes a graph and divides it into a specific number of batches (options to randomize within batches and globally)
- analyze_input.py: counts number of lines in the file to ensure that the batching has happened correctly
- count_vertices_in_batch.py: counts the number of vertices in a given batch
- count_edges_in_batch.py: count the number of edges in a given batch
- generate_plots.py: Creates performance bar charts for scalability comparison (for a single method)

Adds more complete README which specifies more fine-grained goals/tasks in the project

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,71 +23,70 @@ make -j4
...
-Our checks enforce syntax validations and formatting.
-We encourage contributors to use pre-commit hooks.
-
-```shell
-# install all pre-commit hooks
-make hooks
-
-# run pre-commit on repo once
-make pre-commit
-```
```
Comment (meyerzinn) [inline README.md:97]: we are going to want to keep these developer docs ^

**Comment by patrickkenney9801** [comment]
You need to run `make hooks` on the repo and then `make pre-commit` and commit any changes made by the hooks
In the future this will not be necessary since your hooks will be set up

**Comment by meyerzinn** [comment]
@anubhavgoel26 can we get this merged in today?

**Comment by anubhavgoel26** [comment]
> @anubhavgoel26 can we get this merged in today?

Yes I need to make some pre-commit changes, should be good to merge after that.

File: scripts/script.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# SPDX-License-Identifier: BSD-2-Clause
```
Comment (patrickkenney9801) [inline scripts/script.py:1]: please give this file a more descriptive name

File: README.md (hunk 1 above)
Comment (anubhavgoel26) [reply README.md:97]: Passes all pre-commits

File: scripts/script.py (hunk 2 above)
Comment (anubhavgoel26) [reply scripts/script.py]: Made all script names more descriptive

**Review by AdityaAtulTewari — APPROVED** [review]

**Review by meyerzinn — APPROVED** [review]

**Pull request merged**